### PR TITLE
Fuel tank capacities

### LINKED ISSUE #2285: [BUG] Wrong Fuel Tank Capacities
**Mod Version**
master - doesn't matter

**Describe the bug**
The fuel capacities of the various fuel tanks are incorrect. Also incorrect is the amount of unusable fuel in each tank.

**To Reproduce**

1. Compare the fuel tank capacities in either the flight_model.cfg file, or in the weight and balance page of the MSFS flight planner ot the reference information provided below.
2.  The flight_model.cfg specifies "0" for the unusable fuel quantities for each tank, which is incorrect.

**Expected behavior**

**Actual behavior**

**References**

FAA Type Certification Data Sheet:
Screenshot (371)
EASA Type Certification Data Sheet:
Screenshot (373)
A320NEO Airplane Flight Manual:
Screenshot (375)_LI
Screenshot (375)_LI

**Additional context**

Note that the TCDSs and the AFM show 3 fuel tanks, but the FCOM shows the wing fuel tanks divided into inner and outer sections.
Was this working before/when did the issue start occurring?

Is this a problem in the vanilla unmodded game?yes

Discord username (if different from GitHub):donbikes#4084

### PR DESCRIPTION
**Pull request by donstim**
Fixes #2285 

## Summary of Changes
Fixes the incorrect fuel tank capacities used in the sim for the A320NEO.

## References
FAA Type Certificate Data Sheet:
Screenshot (373)

EASA Type Certificate Data Sheet:
Screenshot (371)

A320NEO Airplane Flight Manual:
Screenshot (375)_LI

## Additional context

Discord username (if different from GitHub):donbikes#4084

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by donstim** [comment]
Update:  I just found that the neo FCOM we have been using does divide the wing tank into inner and outer tanks. So, for now, to avoid possible confusion from simmers over the change in number of fuel tanks, I will go with the FCOM and keep the current fuel tank configuration. I don't understand how the FCOM can differ from the AFM and the Type Certificate Data Sheets, but there it is. IF anyone can shine further light on this, I'm all ears.
Screenshot (377)

**Comment by michaeldinozzo** [comment]
@donstim I think it has something to do with the factor they are using to convert liters (or gallons) to kg (or lbs). It is depending on various environmental parameters.
Total fuel capacity seems to be at 23,725 liters in all 3 documents.

**Comment by donstim** [comment]
@michaeldinozzo The question I had actually was not about the total fuel capacity, I know that's consistent between the documents. The thing that is inconsistent is the number of fuel tanks. The TCDS and AFM show a center tank and 2 wing tanks -- 3 tanks in all. The FCOM shows a center tank and inner and outer tanks in each wing for a total of 5 tanks like are currently in the sim.

**Comment by donstim** [comment]
Although the relevant variable names in the flight_model.cfg file are "total fuel capacity" and "unusable fuel capacity," it appears that the "total fuel capacity" is actually the usable fuel capacity, just like it was in FSX. I say this because there is no fuel left in the fuel tanks shown in the MSFS weights and balance page when all the usable fuel is burned. There is also no fuel showing on the ECAM fuel page when the fuel is exhausted. I don't know if the ECAM indication should show the unusable fuel left in the tanks after all usable fuel has been exhausted.

**Comment by beheh** [comment]
This should probably not close #2285 as it doesn't fully fix the wrong number of fuel tanks.

OTOH I think it's fine if we have the sources and you want to make the case in this PR to drop the fuel tanks to three.

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg [hunk 1]
```diff
@@ -114,7 +114,7 @@ Line.29 = Name:Xfer1ToTank2#Source:LeftTransferValve1#Destination:LeftInner#Grav
 Line.30 = Name:Xfer2ToTank3#Source:RightTransferValve1#Destination:RightInner#GravityBasedFuelFlow:500
 Line.31 = Name:Tank4ToXfer1_2#Source:LeftOuter#Destination:LeftTransferValve2#GravityBasedFuelFlow:500
 Line.32 = Name:Tank5ToXfer2_2#Source:RightOuter#Destination:RightTransferValve2#GravityBasedFuelFlow:500
-Line.33 = Name:Xfer1ToTank2_2#Source:LeftTransferValve2#Destination:LeftInner#GravityBasedFuelFlow:500
+Line.23 = Name:Xfer1ToTank2_2#Source:LeftTransferValve2#Destination:LeftInner#GravityBasedFuelFlow:500
```
Comment (lukecologne) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg:117]: This looks like it wasn't intentional

Comment (donstim) [reply A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg:117]: Thanks!

Comment (donstim) [reply A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg]: That was weird.  That typo was not in the branch this pull request was from. I had to change the branch to include the typo, then change it back to fix this.

**Comment by donstim** [comment]
> This should probably not close #2285 as it doesn't fully fix the wrong number of fuel tanks.
> 
> OTOH I think it's fine if we have the sources and you want to make the case in this PR to drop the fuel tanks to three.

Do you want me to revise the issue?  I put the issue together, then found the FCOM material regarding the number of fuel tanks. This PR is to close that issue. I can revise the issue accordingly if need be.

**Comment by ko-zone** [comment]
LGTM

**Comment by beheh** [comment]
>Do you want me to revise the issue? I put the issue together in order to make a PR to address it, then found the FCOM material regarding the number of fuel tanks when completing this PR. This PR is intended to close that issue. I can revise the issue accordingly if need be.

My point is just that you put "Fixes #2285" in your description, which means GitHub will automatically close #2285 as soon as this is merged. But while this issue fixes a part of 2285 it is not completely solved. So I would suggest either removing the "Fixes #2285" (you can write References #2285 or something like that), or to make #2285 only about the quantities and file a new issue for the mismatching number of fuel tanks to fix that in future.

Or alternatively you just fix that remaining part as well as part of this PR :-) Not sure how hard that would be.

**Comment by airfirefighter** [comment]
> @donstim I think it has something to do with the fuel density they are using to convert liters (or gallons) to kg (or lbs). It is depending on various environmental parameters.
> Total fuel capacity seems to be at 23,725 liters in all 3 documents.

This different its because the type of fuel (jetA only in US jetA1 or jetB ect.)as donsitim said deferent density .Iam not a pilot but if i remember  as a firefighter.

**Comment by donstim** [comment]
> My point is just that you put "Fixes #2285" in your description, which means GitHub will automatically close #2285 as soon as this is merged. But while this issue fixes a part of 2285 it is not completely solved. So I would suggest either removing the "Fixes #2285" (you can write References #2285 or something like that), or to make #2285 only about the quantities and file a new issue for the mismatching number of fuel tanks to fix that in future.
> 
> Or alternatively you just fix that remaining part as well as part of this PR :-) Not sure how hard that would be.

I guess I am not making myself clear. It does fully resolve #2285. I found other information since writing #2285 that indicates the number of fuel tanks can be 5 for the neo. See my comment above with a screenshot of the FCOM showing 5 fuel tanks. To resolve your question, I will revise #2285 .

**Comment by beheh** [comment]
Oh! Alright, sorry, I misunderstood this then - I was following along a bit in Discord and the last status I had was "5 tanks are not certified yet", i.e. no real aircraft exists with them. But if that has changed, then yeah, #2285 was just inaccurate before. Thanks!

**Comment by donstim** [comment]
Welllll, as I said in my comment that added the FCOM material, I don't understand how the FCOM can seemingly differ from the cert data sheets and the AFM. Maybe dividing the wing tanks into inner and outer does not actually make it a 5-tank configuration under whatever definition of the number of tanks is used for the cert data sheets and the AFM. Whatever the reason for the discrepancy, I think it would be best to mirror the description in the FCOM. Therefore, I have revised #2285 to remove the number of tanks from being an issue.

**Comment by airfirefighter** [comment]
> Welllll, as I said in my comment that added the FCOM material, I don't understand how the FCOM can seemingly differ from the cert data sheets and the AFM. Maybe dividing the wing tanks into inner and outer does not actually make it a 5-tank configuration under whatever definition of the number of tanks is used for the cert data sheets and the AFM. Whatever the reason for the discrepancy, I think it would be best to mirror the description in the FCOM. Therefore, I have revised #2285 to remove the number of tanks from being an issue.

https://www.youtube.com/watch?v=Cy3HnP6nf0c  from 0:40 5 fuel tanks

**Comment by donstim** [comment]
Thanks for that. I think that pretty much resolves the number of fuel tanks issue even though the videos are for a CEO based on the fuel tank capacity. The capacity in the first video lines up with the basic 3-tank CEO, so I guess 3 tanks in the TCDS and AFM means 5 tanks from a practical and operational standpoint.  It would be interesting to see where the additional tanks are located in a 4 or 5 tank option.

**Comment by airfirefighter** [comment]
> Thanks for that. I think that pretty much resolves the number of fuel tanks issue even though the videos are for a CEO based on the fuel tank capacity. The capacity in the first video lines up with the basic 3-tank CEO, so I guess 3 tanks in the TCDS and AFM means 5 tanks from a practical and operational standpoint. It would be interesting to see where the additional tanks are located in a 4 or 5 tank option.

this is the scheme of the additional 5 tank option https://services.airbus.com/en/flight-operations/system-upgrades/operations-extension/additional-center-tanks-act.html

**Comment by tareksabet** [comment]
so the ACT is an optional thing on the planes but will be added to all the A321Neo XLR, but IMO base yourself from the FCOM as it explains to you the weight and the Volume of each tank. 
Just tell me when ready so I can test it. 
Thx for your great work

**Comment by donstim** [comment]
This PR is based on the FCOM and is ready for testing.

**Comment by tareksabet** [comment]
> This PR is based on the FCOM and is ready for testing. 

Lovely ! 
Will test it as soon as I can

**Comment by donstim** [comment]
Thanks!

**Comment by Awemeter** [comment]
Category: QA Tester
Name: Awemeter#7919
Date of testing: 12/13/2020
Version of the sim: 1.11.7.0 
PR Tested: #2286 
Object of test: Fuel Tank capacities 

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request: Tried out.
1. Loaded into Klas at the gate.
2. Opened the fuel menu. 
3. Moved the slider to 100%.
4. Added all of the fuel tank numbers on a calculator. 


Flight notes (If required):

Issues (If Any): N/A
Overall Rating: Good
Conclusions of the testing: Epic simmer PR looks really nice and adds to the realism. Just cant wait until we don't have to use the garbage fuel menu.

**Comment by Highflyer2010** [comment]
> 
> 
> Category: QA Tester
> Name: Awemeter#7919
> Date of testing: 12/13/2020
> Version of the sim: 1.11.7.0
> PR Tested: #2286
> Object of test: Fuel Tank capacities
> 
> Testing technique: Description of how you tested the Pull Request
> Steps you've made to test the Pull Request: Tried out.
> 
>     1. Loaded into Klas at the gate.
> 
>     2. Opened the fuel menu.
> 
>     3. Moved the slider to 100%.
> 
>     4. Added all of the fuel tank numbers on a calculator. Added up to 6,267 gallons.
> 
> 
> Flight notes (If required):
> 
> Issues (If Any): N/A
> Overall Rating: Good
> Conclusions of the testing: Epic simmer PR looks really nice and adds to the realism. Just cant wait until we don't have to use the garbage fuel menu.

@Awemeter are they going to implement doing it from the MCDU?

**Comment by ImenesFBW** [comment]
Name: Imenes#8739
Date of testing: Current date 14/12/2020
Version of the sim: 1.11.7.0
PR Tested: #2286
Object of test: Checking fuel-tank capacity
Testing technique: Try-out (multiple tests)

## Steps you've made to test the Pull Request:

 * Moved the fuel slider to 100% in the pre-flight menu
 * Flew a full 1h flight from C&D, checked the fuel quantity (6267 Gallons), and kept an eye on possible bugs  
 * Loaded another flight, this time I moved the fuel slider to 100% in-game, the amount was the same (6267 Gallons).

## Overall Rating: Good

#### Conclusions of the testing:
 * Working as documented, thank you for your work! LGTM

**Comment by Benjozork** [comment]
Waiting on @tareksabet for accuracy check

**Review by Benjozork — APPROVED** [review]

**Comment by donstim** [comment]
Somehow missed updating the change log, so that has now been added.

**Comment by tareksabet** [comment]
Name: tarek_sabet#7946
Date of testing: Current date 15/12/2020
Version of the sim: 1.11.7.0
PR Tested: #2286
Object of test: Checking fuel-tank capacity
Testing technique: Try-out (multiple tests)

Steps you've made to test the Pull Request:
Moved the fuel slider to 100% in the pre-flight menu
Flew a  flight from C&D, checked the fuel quantity (6600 Kgs), and checked if anything is wrong.
Loaded another flight, this time I moved the fuel slider to 100% in-game, the amount was the same (18960 Kgs).
Overall Rating: Good
Conclusions of the testing:
appeared as from the FCOM everything is correct. Very good job. LGTM

**Comment by markuskorbel** [comment]
I agree that these values should be realistic, one wee gotha though: addons like onair are now detecting this as a new/incompatible aircraft to the default asobo 320neo. I'm currently recording a demo flight to get it added to onair - not sure if I can convert my existing planes to the new one...

**Pull request merged**